### PR TITLE
Add shell checker

### PR DESCRIPTION
**Pull request by rollandf**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  shellcheck:
+    name: Shellcheck
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run ShellCheck
+        uses: ludeeus/action-shellcheck@master
+        with:
+          severity: error
```
Comment (adrianchiris) [inline .github/workflows/test.yml:19]: nit: missing newline

**Review by adrianchiris — COMMENTED** [review]
@rollandf anything missing from this PR ?

we got lots of issues that need fixing as a result of this check ?

**Comment by rollandf** [comment]
> @rollandf anything missing from this PR ?
> 
> we got lots of issues that need fixing as a result of this check ?

Yes, there are a few

**Review by adrianchiris — APPROVED** [review]
LGTM!

**Review by e0ne — APPROVED** [review]

**Pull request merged**